reference/concepts.md

```diff
@@ -87,7 +87,7 @@ The Haskell concept exercises are based on concepts. The list below contains the
   - [ ] Queues
   - [ ] Sets
   - [ ] Stacks
-- [ ] Algebraic data types
+- [x] Algebraic data types
```

Comment:
I can't agree that we've covered these. I would have had to see an exercise explaining `data`, but I don't see one

Reply:
ADTs and `data` are WIP under #1110. The idea being PR #1110 will be finished prior to this one.

Reply:
Well since I intend to merge this now, before #1110, I'm removing this change. Resubmit it after #1110 is done, or just include this change in #1110.